File: CS/PS/Site/leetcode/graph/77.2.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number[][]}
 */
 var combine = function(n, k) {
   let lst = Array(n).fill(0).map((_,idx) => idx + 1);
   const result = []
   function dfs(tmp, start){
     if (tmp.length === k){
       result.push([...tmp]);
       return;
     }
     for(let i = start; i < n; i++){
       tmp.push(lst[i]);
       dfs(tmp, i + 1);
       tmp.pop();
     }
   }
   dfs([], 0);
   return result
};

console.log(combine(4, 2));

File: CS/PS/Site/leetcode/graph/77.2.py
from typing import List


class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        lst = [v for v in range(1, n + 1)]
        result = []
        tmp = []
        print(lst)
        def dfs(tmp, start):
          if len(tmp) == k:
            result.append(tmp[:])
            return
          for i in range(start, n):
            tmp.append(lst[i])
            dfs(tmp, i + 1)
            tmp.pop()
        dfs(tmp, 0)
        return result
    print(combine("", 4, 2))
